Output separate YAML files for cisagov conversion

mcdonnnj opened the pull request:
# <!-- Use the title to describe PR changes in the imperative mood --> #

## 🗣 Description ##

This pull request changes the `convert-cisagov` script from printing the YAML for the convert Markdown to `stdout` to instead output a series of files in a pre-defined format. These files contain entries grouped by the first character (in uppercase) of the product vendor.
<!-- Describe the "what" of your changes in detail. -->
<!-- To avoid scope creep, limit changes to a single goal. -->

## 💭 Motivation and context ##

The resultant YAML file from converting the existing Markdown file is too large for the GitHub web UI to work with. There is an API limitation of 1MiB that a file needs to stay under for the web UI to allow in-browser editing/viewing. When the converted YAML is distributed across multiple files we can keep the individual file size to well below the 1MiB limit.
<!-- Why is this change required? -->
<!-- What problem does this change solve? How did you solve it? -->
<!-- Mention any related issue(s) here using appropriate keywords such -->
<!-- as "closes" or "resolves" to auto-close them on merge. -->

## 🧪 Testing ##

Automated tests pass. I confirmed that the processed Markdown is split up in the specified way when run locally.
<!-- How did you test your changes? How could someone else test this PR? -->
<!-- Include details of your testing environment, and the tests you ran to -->
<!-- see how your change affects other areas of the code, etc. -->

<!--
## 📷 Screenshots (if appropriate) ##

Uncomment this section if a screenshot is needed.

-->

## ✅ Pre-approval checklist ##

<!-- Remove any of the following that do not apply. -->
<!-- Draft PRs should have one or more unchecked boxes. -->
<!-- If you're unsure about any of these, don't hesitate to ask. -->
<!-- We're here to help! -->

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - _eschew scope creep!_
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All new and existing tests pass.

mcdonnnj committed:
Output separate YAML files for cisagov conversion

Instead of printing one large YAML file to stdout we instead group the
read software entires by the uppercased first character of the vendor
name and output a file for each of these groups. All non-alphabet
vendor entries are grouped in the "0-9" group manually. When testing
the only entries that were non-alphabet were numeric. This will generate
files that are small enough to be edited using the GitHub web UI.

jsf9k commented on src/mdyml/convert_cisagov.py:
```diff
@@ -160,15 +162,31 @@ def convert() -> None:
+        if not key.isalpha():
+            non_letter_groups.extend(out_dict_groups[key])
+            del out_dict_groups[key]
+    out_dict_groups["0-9"] = non_letter_groups
```
Are there any funky cases where the leading character of a vendor name is not alphanumeric?  It might be better to name this group `out_dict_groups["nonalpha"]` since you are really only checking that.

mcdonnnj replied:
Nothing in our data contained that, but yes that is a case we should consider. I was mainly mirroring how the data is split up in https://github.com/NCSC-NL/log4shell/tree/main/software honestly.

jsf9k replied:
Given that this is international and supports Unicode, there could be some oddballs in there.

This is a good time to remind @felddy to meditate upon [The Solidus](https://unicode-search.net/unicode-namesearch.pl?term=SOLIDUS).

mcdonnnj replied:
Speaking of Unicode we should _probably_ be normalizing the mapping keys...

dav3r replied:
I agree with @jsf9k and prefer `"nonalpha"` to `"0-9"` here.

mcdonnnj replied:
Please see d42b79b for your consideration.

jsf9k reviewed (approved):
Approved, but please consider my one suggestion.

dav3r reviewed (approved):
Very nice! 🤩

mcdonnnj committed:
Update the key for non-alphabet groups

Use a more general (and accurate) key for the group of vendors that
begin with a non-alphabetic character.

Co-authored-by: Shane Frasier <jeremy.frasier@trio.dhs.gov>
Co-authored-by: dav3r <david.redmin@trio.dhs.gov>
